### PR TITLE
Influx: use github.com/influxdata/line-protocol/v2 Encoder instead of strings

### PR DESCRIPTION
**Pull request by tdanner**
I am still hitting edge cases with influx encoding. Rather than continue to read https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/ ever more closely and keep tweaking the string handling in influx.go, I think it would be better to switch to the "official" encoder.

Right now I have it checking the error state of the encoder after each step and panicking if it is non-nil. This is quite verbose and, while helpful during debugging, may not be how we want to handle errors in production. Feedback welcome.

### CODE REVIEW

**Review by i3149 — COMMENTED** [review]
I'm personally not a fan of panic statements in agent code. Since its so hard to get customers to run it/update it, better to make agents as bulletproof as possible. Rather than panicking on each error, I would consider logging an error message once and then updating an internal metric with the error count. 

Downside of this is that we're not currently passing the metrics registry into the format block, but this should be easy enough to do.

**Comment by tdanner** [comment]
Sounds good - I'll plumb the telemetry stuff into formatters and log errors instead of calling panic.

File: pkg/formats/influx/influx.go [hunk 1]
```diff
@@ -117,7 +91,16 @@ func NewMergedInfluxData(s InfluxDataSet) *InfluxData {
 
 type InfluxDataSet []InfluxData
 
-func (s InfluxDataSet) Bytes() []byte {
+func (f *InfluxFormat) report(err error, format string, args ...interface{}) {
+	str := fmt.Sprintf(format, args...)
+	if !f.invalids[str] {
```
Comment (i3149) [inline pkg/formats/influx/influx.go:96]: There's a small chance this will panic with a concurrent map access. Consider wrapping into a sync.RWMutex?

**Review by i3149 — APPROVED** [review]
Looks awesome! Probably worth adding a mutex just because if there's a ton of traffic which is also erroring you'll probably hit a panic.

File: pkg/formats/influx/influx.go (hunk 1 above)
Comment (tdanner) [reply pkg/formats/influx/influx.go:96]: Good point. Will do.

**Review by i3149 — APPROVED** [review]
Ship it!

**Pull request merged**